### DISCUSSION TITLE
Problem adding NR entity nodes to existing subflows

### DISCUSSION THREAD

**Discussion by inguy24** [zachowj/hass-node-red#275] [Q&A]
Hi, I am having an issue when I add a new entity node to an existing Node Red subflow. What happens is that entity node (i.e. sensor, switch etc) when registered with HA, gets duplicated the same number of times that the subflow appears in  other flows. For example, when creating a new switch entity node in a subflow that may be called multiple times from the same flow... I get 6 switch entities registered in HA (i.e. switch_1 switch_2, etc.) If I create the entity node in a regular flow, and then create the subflow by selecting those nodes, then the duplication does not happen. What am I doing wrong here, or is this a bug that needs addressed? Although I try not to create subflows first, before adding entity nodes, sometimes it happens with regular tweaking or building out of more complicated flows.

**Comment by zachowj** [comment]
What version of this component and [Home Assistant nodes](https://github.com/zachowj/node-red-contrib-home-assistant-websocket) do you have installed?

<https://zachowj.github.io/node-red-contrib-home-assistant-websocket/guide/diagnostics.html>

**Comment by inguy24** [comment]
Companion is 4.0.0
websocket is 0.64

**Comment by zachowj** [comment]
Oh, that's interesting. It looks to be a bug with node-RED. I don't believe that is the expected behavior. I'll have to take a closer look.

**Comment by zachowj** [comment]
Not a bug with node-RED. It's a scoping issue with the entity config node.

Edit the entity config scope to "On all flows" and it should stop the duplication.

![image](https://github.com/zachowj/hass-node-red/assets/37859597/616a0b77-3e57-4e99-909c-2205d8521fb4)

**Comment by inguy24** [comment]
Hmmm that is it's default setting. I double checked just in case. Again it only happens if I initially create a new entity node in an existing subflow but not if I create it then create the subflow with move selection to subflow. Unfortunately I cannot delete the additiona HA entities as they are all still registered to node red companion. Node red does not show these as additional configuration nodes... just the original node.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by zachowj** [reply]
You would have to edit the entity config from within the subflow. Edit the subflow, click on the switch node, edit the entity config, and change its scope.

Config nodes that are in the scope of a subflow don't show in the config sidebar. It only shows hidden by the subflow.
